### PR TITLE
fix: eth: return the correct nonce from EthGetTransactionCount

### PR DESCRIPTION
**Pull request by Stebalien**
## Related Issues

fixes https://github.com/filecoin-project/lotus/issues/10255

## Proposed Changes

In EthGetTransactionCount, get the "correct" nonce for EVM contracts in case someone is using this method to, e.g., predict CREATE addresses.

EVM contracts track this number internally.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

File: chain/actors/builtin/evm/evm.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ type State interface {
 	cbor.Marshaler
 
 	Nonce() (uint64, error)
+	IsAlive() (bool, error)
```
Comment (shrenujbansal) [inline chain/actors/builtin/evm/evm.go:51]: I don't see an implementation anywhere? Is this the right PR?

File: node/impl/full/eth.go [hunk 2]
```diff
@@ -367,6 +368,29 @@ func (a *EthModule) EthGetTransactionCount(ctx context.Context, sender ethtypes.
 		return ethtypes.EthUint64(0), xerrors.Errorf("cannot parse block param: %s", blkParam)
 	}
 
+	// First, handle the case where the "sender" is an EVM actor.
```
Comment (shrenujbansal) [inline node/impl/full/eth.go:371]: this comment should be on line 380?

File: chain/actors/builtin/evm/evm.go (hunk 1 above)
Comment (shrenujbansal) [reply chain/actors/builtin/evm/evm.go:51]: nvm

File: node/impl/full/eth.go (hunk 2 above)
Comment (Stebalien) [reply node/impl/full/eth.go:371]: It's the entire section. I've put it into a block to make that clear.

**Review by shrenujbansal — APPROVED** [review]

File: node/impl/full/eth.go (hunk 2 above)
Comment (Stebalien) [reply node/impl/full/eth.go:371]: Eh, I switched it to an if chain.

**Review by arajasek — APPROVED** [review]

**Pull request merged**